pkg/server/testing/fake_execution_client.go

```diff
@@ -0,0 +1,315 @@
+	defer f.Unlock()
+	names := []string{}
+	for _, detail := range f.called {
+		names = append(names, detail.name)
+	}
+	return names
+}
+
+// Create is a test implementation of execution.Create.
+func (f *FakeExecutionClient) Create(ctx context.Context, createOpts *execution.CreateRequest, opts ...grpc.CallOption) (*execution.CreateResponse, error) {
+	f.Lock()
+	defer f.Unlock()
+	f.appendCalled("create", createOpts)
+	if err := f.popError("create"); err != nil {
+		return nil, err
+	}
+	_, ok := f.ContainerList[createOpts.ID]
+	if ok {
+		return nil, containerd.ErrContainerExists
+	}
+	pid := generatePid()
+	f.ContainerList[createOpts.ID] = &container.Container{
+		ID:     createOpts.ID,
+		Pid:    pid,
+		Status: container.Status_CREATED,
+	}
+	f.sendEvent(&container.Event{
+		ID:   createOpts.ID,
+		Type: container.Event_CREATE,
+		Pid:  pid,
+	})
+	return &execution.CreateResponse{
+		ID:  createOpts.ID,
+		Pid: pid,
+	}, nil
+}
+
+// Start is a test implementation of execution.Start
+func (f *FakeExecutionClient) Start(ctx context.Context, startOpts *execution.StartRequest, opts ...grpc.CallOption) (*google_protobuf.Empty, error) {
+	f.Lock()
+	defer f.Unlock()
+	f.appendCalled("start", startOpts)
+	if err := f.popError("start"); err != nil {
+		return nil, err
+	}
+	c, ok := f.ContainerList[startOpts.ID]
+	if !ok {
+		return nil, containerd.ErrContainerNotExist
+	}
+	f.sendEvent(&container.Event{
+		ID:   c.ID,
+		Type: container.Event_START,
+		Pid:  c.Pid,
+	})
+	switch c.Status {
+	case container.Status_CREATED:
+		c.Status = container.Status_RUNNING
+		return &google_protobuf.Empty{}, nil
+	case container.Status_STOPPED:
+		return &google_protobuf.Empty{}, fmt.Errorf("cannot start a container that has stopped")
+	case container.Status_RUNNING:
+		return &google_protobuf.Empty{}, fmt.Errorf("cannot start an already running container")
+	default:
+		return &google_protobuf.Empty{}, fmt.Errorf("cannot start a container in the %s state", c.Status)
+	}
+}
+
+// Delete is a test implementation of execution.Delete
+func (f *FakeExecutionClient) Delete(ctx context.Context, deleteOpts *execution.DeleteRequest, opts ...grpc.CallOption) (*execution.DeleteResponse, error) {
+	f.Lock()
+	defer f.Unlock()
+	f.appendCalled("delete", deleteOpts)
+	if err := f.popError("delete"); err != nil {
+		return nil, err
+	}
+	c, ok := f.ContainerList[deleteOpts.ID]
+	if !ok {
+		return nil, containerd.ErrContainerNotExist
+	}
+	delete(f.ContainerList, deleteOpts.ID)
+	f.sendEvent(&container.Event{
+		ID:   c.ID,
+		Type: container.Event_EXIT,
+		Pid:  c.Pid,
+	})
+	return nil, nil
+}
+
+// Info is a test implementation of execution.Info
+func (f *FakeExecutionClient) Info(ctx context.Context, infoOpts *execution.InfoRequest, opts ...grpc.CallOption) (*container.Container, error) {
+	f.Lock()
+	defer f.Unlock()
+	f.appendCalled("info", infoOpts)
+	if err := f.popError("info"); err != nil {
+		return nil, err
+	}
+	c, ok := f.ContainerList[infoOpts.ID]
```

Comment:
Same problem here. How about just change `ContainerList` to `map[string]container.Container`? We don't actually need a pointer.